main.rb

```diff
@@ -0,0 +1,149 @@
+# rubocop: disable Style/For, Style/CaseEquality, Metrics/PerceivedComplexity, Metrics/MethodLength
+# rubocop: disable Style/RedundantSelf, Metrics/CyclomaticComplexity, Metrics/ModuleLength
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    for i in self do
+      yield(i)
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    my_arr = to_a
+    len = self.length
+    until i == len
+      yield(my_arr[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    arr = []
+    for i in self do
+      arr.push(i) if yield(i)
+    end
+    arr
+  end
+
+  def my_all?(par1 = nil)
+    if par1
+      for i in 0...length do
+        return false unless par1 === self[i]
+      end
+    elsif block_given?
+      for i in self do
+        return false unless yield(i)
+      end
+    else
+      for i in self do
+        return false if i.nil? || i === false
+      end
+    end
+    true
+  end
+
+  def my_any?(par1 = nil)
+    if par1
+      for i in 0...length do
+        return true if par1 === self[i]
+      end
+    elsif block_given?
+      for i in self do
+        return true if yield(i)
+      end
+    else
+      for i in self do
+        return true if i
+      end
+    end
+    false
+  end
+
+  def my_none?(par1 = nil)
+    if par1
+      for i in 0...length do
+        return false if par1 === self[i]
+      end
+    elsif block_given?
+      for i in self do
+        return false if yield(i)
+      end
+    else
+      for i in self do
+        return false if i
+      end
+    end
+    true
+  end
```

Comment:
[OPTIONAL]

I think we could get the block of my_none? from my_any? we should only do something like !my_any? for my_none? to work. For example:

```ruby
def my_none?(arg = nil, &block)
!my_any?(arg, &block)
end
```

To make this work we have to use block.call(element) instead of yield(element) in my_any?